### DISCUSSION TITLE
How to fetch Cron Trigger list by filtering arguments,  Also rest-apis for cron pending events, processed events and invocationLog

### DISCUSSION THREAD

**Discussion by SameerChorge94** [hasura/graphql-engine#8284] [Question]
Hello ,

I Checked hasura metadata api for creating, deleting, Fetching  the CRON triggers 
while fetching the CRON triggers list 
is there any way to get the filtered list  based on arguments options provided OR fetch by name 

> URL:http://localhost:8083/v1/metadata
> Req. Body: 
```
{
    "type" : "get_cron_triggers",
    "args" : {
        
    }
}

```

I also tried by providing the name variable inside args json but it's still returning the complete list 
```
{
    "type" : "get_cron_triggers",
    "args" : {
        "name":"Cron_Using_Medata_API_test"
    }
}
```
**is there any way to fetch this data filtered?**

Also as per our project requirement we also need to **fetch the data for Pending Events, Processed Events and Invocation Logs if possible**

**Comment by SameerChorge94** [comment]
For cron trigger event list filter refer this discord message link 
https://discord.com/channels/407792526867693568/428469959530643466/950705792661721088

![image](https://user-images.githubusercontent.com/55871576/157224382-df3f99cc-08c9-41ba-86e7-fc16b3347e49.png)
it stated that 
> at the moment seems to only accept type and event_id as filters
I'm sure it's a valid feature request to be able to filter by everything though


https://hasura.io/docs/latest/graphql/core/api-reference/metadata-api/scheduled-triggers.html#get-cron-triggers

for the **fetch the data for Pending Events, Processed Events and Invocation Logs using metadata api** 
i didn't get any official documentation, instead i got called api from hasura UI from network section by inspecting browser window
here are the list of payloads for used for those apis


## Hasura metadata Rest-API's details for Cron Triggers:

> Hasura Metadata URl: http://<HASURA_URL>/<version>/metadata

> E.g. URL : http://localhost:8083/v1/metadata
###### 1. For creating Cron triggers using Rest-API 
Ref: https://hasura.io/docs/latest/graphql/core/api-reference/metadata-api/scheduled-triggers.html
```
{
    "type": "create_cron_trigger",
    "args": {
        "name": "test",
        "webhook": "http://192.168.0.113:4008/testCronJob1",
        "schedule": "0 0 * * */1",
        "payload": {
            "key1": "value1",
            "key2": "value2"
        },
        "include_in_metadata": true,
        "comment": "sample_cron commment",
        "replace": true,
        "retry_conf": {
            "num_retries": 0,
            "timeout_seconds": 180,
            "tolerance_seconds": 21600,
            "retry_interval_seconds": 10
        }
    }
}
```


###### 2. Get All CRON trigger list
```
{
    "type": "get_cron_triggers",
    "args": {}
}
```
###### 3. Get Pending CRON events (scheduled cron jobs with next occurrence details )
```
{
    "type": "get_scheduled_events",
    "args": {
        "type": "cron",
        "trigger_name": "test",
        "status": [
            "scheduled"
        ],
        "limit": 10,
        "offset": 0
    }
}   
```

###### 4. Get Processed CRON events list
```
{
    "type": "get_scheduled_events",
    "args": {
        "type": "cron",
        "trigger_name": "test",
        "status": ["delivered", "dead", "error"],
        "limit": 10,
        "offset": 0
    }
}   
```

###### 5. Get CRON process invocation logs list
```
{
    "type": "get_event_invocations",
    "args": {
        "type": "cron",
        "trigger_name": "Cron_Using_Medata_API_test",
        "limit": 10,
        "offset": 0
    }
}   
```


###### 6. Delete CRON process 
Ref: https://hasura.io/docs/latest/graphql/core/api-reference/metadata-api/scheduled-triggers.html#delete-cron-trigger
```
{
    "type" : "delete_cron_trigger",
    "args" : {
        "name": "test"
    }
}
```

**Comment by codingkarthik** [comment]
Hi @SameerChorge94,

Thanks for creating this discussion, so a couple of arguments I think would be helpful for filtering the cron triggers list would be:

1. filter by `include_in_metadata`
2. filter by `names` (array of names)
